packages/next/src/server/config-shared.ts

```diff
@@ -23,20 +23,20 @@ export type NextConfigComplete = Required<NextConfig> & {
   configFileName: string
 }
 
-export type I18NDomains = DomainLocale[]
+export type I18NDomains = readonly DomainLocale[]
 
 export interface I18NConfig {
   defaultLocale: string
   domains?: I18NDomains
   localeDetection?: false
-  locales: string[]
+  locales: readonly string[]
```

Comment:
I needed to double-check, so I'm adding it here for posterity: A change in this direction (adding `readonly`) will not lead to compiler errors in users' existing Next.js configs, only the other direction (removing `readonly`) would potentially be problematic.